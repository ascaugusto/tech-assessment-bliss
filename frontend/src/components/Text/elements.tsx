import styled from 'styled-components'
import { COLORS } from '../../utils/constants'

interface textComponentProps {
  size?: string,
  color?: string,
  clicker?: boolean,
  bold?: boolean,
  paddingRight?: string
}

export const TextComponent = styled.div<textComponentProps>`
  font-size: ${props => props.size ? props.size : '12px'};
  color: ${props => props.color ? props.color : COLORS.pearlAqua};
  ${props => props.clicker && 'cursor: pointer;' }
  ${props => props.bold && 'font-weight: bold;' }
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  ${props => props.paddingRight && `padding-right: ${props.paddingRight};` }
`