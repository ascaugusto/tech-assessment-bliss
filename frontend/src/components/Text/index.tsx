import React from 'react'
import { TextComponent } from './elements'

type childrenType = {
  children: string,
  size?: string,
  color?: string,
  clicker?: boolean,
  bold?: boolean,
	paddingRight?: string
}

function Text({ children, size, color, clicker, bold, paddingRight}: childrenType) {
	return (
		<TextComponent
			size={size}
			color={color}
			clicker={clicker}
			bold={bold}
			paddingRight={paddingRight}
		>
			{children}
		</TextComponent>
	)
}

export default Text