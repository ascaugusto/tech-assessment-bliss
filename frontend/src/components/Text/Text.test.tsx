import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Text from '.'

test('<Text />', () => {
	const tree = renderer.create(
		<Text>
        Teste
		</Text>).toJSON()
	expect(tree).toMatchSnapshot()
})