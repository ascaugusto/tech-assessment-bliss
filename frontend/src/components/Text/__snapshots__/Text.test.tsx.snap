// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> 1`] = `
<text
  className="sc-bcXHqe dvHPzA"
>
  Teste
</text>
`;
